service/history/execution/mutable_state_builder.go

```diff
@@ -1591,6 +1591,11 @@ func (e *mutableStateBuilder) SetHistorySize(size int64) {
 	e.executionStats.HistorySize = size
 }
 
+func (e *mutableStateBuilder) GetEstimatedMutableStateSize() int {
```

Comment:
this is going to be the major contributor to the workflow execution cache entry sizes. as long as this is accurate enough we should be good.